scripts/deis.sh

```diff
@@ -3,12 +3,24 @@
 deis_healthcheck() {
   wait-for-all-pods "deis"
   local successes
+  local failures
+  local max_attempts=10
   echo "Checking to see if the workflow has come up properly."
-  while [[ ${successes} -lt 10 ]]; do
+  while [[ ${successes} -lt "${max_attempts}" ]] && [[ ${failures} -lt "${max_attempts}" ]]; do
     wait-for-router
-    let successes+=1
-    if ! ((successes % 10)); then
+    if [ $? -eq 0 ]; then
+      let successes+=1
+    else
+      let failures+=1
+    fi
+
+    if ! ((successes % "${max_attempts}")); then
```

Comment:
This feels counterintuitive to me, but I don't really know bash. This is checking that the remainder is `1`, correct, so that 9 attempts would pass?